mutable_tree.go

```diff
@@ -552,8 +570,8 @@ func (tree *MutableTree) balance(node *Node) (newSelf *Node, orphaned []*Node) {
 
 func (tree *MutableTree) addOrphans(orphans []*Node) {
 	for _, node := range orphans {
-		if !node.persisted {
-			// We don't need to orphan nodes that were never persisted.
+		if !node.saved {
```

Comment:
is this if statement necessary?

Reply:
I'm not sure it is even possible to enter that conditional but i just replicated the previous behavior. I would feel more comfortable taking it out after i've fully finished testing. Will leave unresolved for now.